### PR TITLE
Add styling for Newspack Ads

### PR DESCRIPTION
**Pull request by claudiulodro**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR adds styling for Newspack Ads so that they display nicely on the theme and with style packs. 

Screenshots in details.

Above header:
<details>

Screen Shot 2019-08-14 at 8 12 42 AM
Screen Shot 2019-08-14 at 8 13 12 AM
Screen Shot 2019-08-14 at 8 13 34 AM
Screen Shot 2019-08-14 at 8 14 25 AM

</details>

Below header:
<details>

Screen Shot 2019-08-14 at 8 18 05 AM
Screen Shot 2019-08-14 at 8 18 45 AM
Screen Shot 2019-08-14 at 8 19 03 AM
Screen Shot 2019-08-14 at 8 19 21 AM

</details>

Above footer:
<details>

Screen Shot 2019-08-14 at 8 50 18 AM
Screen Shot 2019-08-14 at 8 51 06 AM

</details>

### How to test the changes in this Pull Request:

1. Install Newspack Ads, create an ad using one of our sample codes. Set it as an above header/below header/above footer ad in the Newspack > Advertising > Global Settings.
2. Try out the different style packs and header settings with the ad.
3. Go to step 1.

To the best of my knowledge, responsiveness of the ads should be handled by the ad itself. We can't make a 900px wide ad responsive on our end since its within an iframe.
 
### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by laurelfulford** [comment]
A breakdown of testing steps for novices can be found here (thanks Claudiu!): p1565803814199700-slack-newspack-private

File: sass/plugins/newspack-ads.scss [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+				margin-bottom: #{ 0.7 * $size__spacing-unit }
+			}
+		}
+	}
+
+	// Style pack 2 can't really support an ad below the header because the content is inset into the header.
+	&.style-pack-style-2 {
+		.newspack_global_ad.global_below_header {
+			display: none;
+		}
```
Comment (laurelfulford) [inline sass/plugins/newspack-ads.scss]: I noticed in some cases there's a lot of space below the 'below header' ad. 

image

This is coming from the theme -- it looks like I need to do some broader reworking of styles to fix it on the front page, but adding something like this inside the `body {}` selector should take care of it in most spots:

```
       // Reduce space on top of the content a bit when there's an ad below the header.
        &:not(.newspack-front-page) .newspack_global_ad.global_below_header + .site-content {
               margin-top: #{ 1.5 * $size__spacing-unit };
        }
```

File: sass/plugins/newspack-ads.scss [hunk 2]
```diff
@@ -0,0 +1,37 @@
+@import "sass/variables-site/variables-site";
+@import "sass/mixins/mixins-master";
+
+.newspack_global_ad {
+	display: flex;
+	justify-content: center;
+	align-items: center;
+
+	> div {
```
Comment (laurelfulford) [inline sass/plugins/newspack-ads.scss]: When AMP is enabled, I noticed it doesn't have the `div` in the markup, so you lose this styling:

image

Switching this from `> div` to `> *` will make it select whatever the first child is, or `> div, > amp-ad` to get the tag that AMP's wrapping it in.

Comment (claudiulodro) [reply sass/plugins/newspack-ads.scss]: Good catch! Fixed in https://github.com/Automattic/newspack-theme/pull/247/commits/928fb0f3b5387965647a973f7b7ba24ae58d22bf

File: sass/plugins/newspack-ads.scss (hunk 1 above)
Comment (claudiulodro) [reply sass/plugins/newspack-ads.scss]: 👍 Much better. Added in https://github.com/Automattic/newspack-theme/pull/247/commits/928fb0f3b5387965647a973f7b7ba24ae58d22bf

**Comment by claudiulodro** [comment]
Thanks for the review! I've made the requested changes. I've also switched the dark gray ad background to the light gray ad background for default-height, default-color header, since it appears the dark gray line that used to be at the top of the header is now gone:
Screen Shot 2019-08-15 at 8 25 05 AM

**Comment by laurelfulford** [comment]
Thanks @claudiulodro - the original issues are all resolved! 

There's one more case I missed for that below-the-header ad space: even though it's hidden with Style C, the markup is still there -- so that style I gave to fix up the spacing in other style packs messes it up 🤦‍♀ 

Changing this style:

``` 
	// Reduce space on top of the content a bit when there's an ad below the header.
	&:not(.newspack-front-page) .newspack_global_ad.global_below_header + .site-content {
		margin-top: #{ 1.5 * $size__spacing-unit };
	}
```

to:

```
	// Reduce space on top of the content a bit when there's an ad below the header.
	&:not(.newspack-front-page):not(.style-pack-style-2) .newspack_global_ad.global_below_header + .site-content {
		margin-top: #{ 1.5 * $size__spacing-unit };
	}
```
... should fix it.

**Before:** 

image

**After:**

image

Sorry about that! I can push up a fix for that if it's easier -- just let me know.

**Comment by laurelfulford** [comment]
After touching base with @claudiulodro, I fixed the above issue in c236196. 

Everything else looks good to me!

**Pull request merged**